projects/observability/src/shared/components/topology/d3/d3-topology.ts

```diff
@@ -420,6 +429,97 @@ export class D3Topology implements Topology {
     }
   }
 
+  private checkAndHandleGroupNodeClick(node: RenderableTopologyNode): void {
+    const userNode = node.userNode;
+
+    if (!isTopologyGroupNode(userNode)) {
+      return;
+    }
+
+    const childNodes = userNode.children;
+    this.runAndDrainCallbacks(this.dataClearCallbacks);
+    userNode.expanded = !userNode.expanded;
+
+    if (!userNode.expanded) {
+      this.userNodes = this.userNodes.filter(n => !childNodes.includes(n));
+    } else {
+      this.userNodes = [this.userNodes, childNodes].flat();
+    }
+
+    this.topologyData = this.topologyConverter.convertTopology(
```

Comment:
This re-init should not be part of this method. Or we should make it more dynamic else there will be perf hits.